### PR TITLE
Tidy up the token checking

### PR DESCRIPTION
**Pull request by JamesSmartCell**
Small PR to clean up some untidy code.

- Tidy up code
- Remove redundant methods
- Only check unknown tokens at startup

### CODE REVIEW

File: app/src/main/java/com/alphawallet/app/viewmodel/WalletViewModel.java [hunk 1]
```diff
@@ -315,15 +312,21 @@ private void onFetchTokensCompletable()
...
             balanceTimerDisposable = Observable.interval(0, BALANCE_CHECK_INTERVAL_MILLIS, TimeUnit.MILLISECONDS)
                     .doOnNext(l -> checkBalances()).subscribe();
         }
     }
 
+    private void AddUnresolvedContracts()
+    {
+        Observable.fromArray(ethereumNetworkRepository.getAllKnownContracts(tokensService.getNetworkFilters()).toArray(new ContractResult[0]))
+                .filter(result -> tokensService.getToken(result.chainId, result.name) == null)
+                .forEach(r -> unknownAddresses.add(r)).isDisposed();
```
Comment (hboon) [inline app/src/main/java/com/alphawallet/app/viewmodel/WalletViewModel.java]: Is the `isDisposed()` at the end necessary?

File: app/src/main/java/com/alphawallet/app/viewmodel/WalletViewModel.java [hunk 2]
```diff
@@ -315,15 +312,21 @@ private void onFetchTokensCompletable()
...
-        checkUnknownAddresses();
+        AddUnresolvedContracts();
         if (balanceTimerDisposable == null || balanceTimerDisposable.isDisposed())
         {
             balanceTimerDisposable = Observable.interval(0, BALANCE_CHECK_INTERVAL_MILLIS, TimeUnit.MILLISECONDS)
                     .doOnNext(l -> checkBalances()).subscribe();
         }
     }
 
+    private void AddUnresolvedContracts()
```
Comment (hboon) [inline app/src/main/java/com/alphawallet/app/viewmodel/WalletViewModel.java]: Matter of style. Start `AddUnresolvedContracts()` with a lowercase instead?

File: app/src/main/java/com/alphawallet/app/viewmodel/WalletViewModel.java (hunk 1 above)
Comment (JamesSmartCell) [reply app/src/main/java/com/alphawallet/app/viewmodel/WalletViewModel.java]: Nicely spotted. It's not necessary, but it's a harmless way to stop Java lint from complaining that the return is not used - which we never need to use in these cases.

File: app/src/main/java/com/alphawallet/app/viewmodel/WalletViewModel.java (hunk 2 above)
Comment (JamesSmartCell) [reply app/src/main/java/com/alphawallet/app/viewmodel/WalletViewModel.java]: Yes probably.

**Pull request merged**